src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,121 @@
 package core.basesyntax;
 
-public class MyHashMap<K, V> implements MyMap<K, V> {
+public class MyHashMap<K,V> implements MyMap<K,V> {
+    private static final int DEFAULT_ARRAY_LENGTH = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private int elementsAmount;
+    private Node<K,V> [] valuesArray;
+    private int arrayIndex;
 
     @Override
     public void put(K key, V value) {
-
+        if (elementsAmount == 0) {
+            valuesArray = new Node[DEFAULT_ARRAY_LENGTH];
+        }
+        if (elementsAmount != 0 && elementsAmount > (valuesArray.length * LOAD_FACTOR)) {
+            resize();
+        }
+        arrayIndex = (key == null ? 0 : Math.abs(key.hashCode() % valuesArray.length));
+        if (valuesArray[arrayIndex] == null) {
+            Node<K, V> newNode = new Node<>(key, value, null);
+            valuesArray[arrayIndex] = newNode;
+            elementsAmount++;
+        } else {
+            Node<K,V> currentNode = valuesArray[arrayIndex];
+            while (true) {
+                if (currentNode.nodeKey != null && currentNode.nodeKey.equals(key)
+                        || currentNode.nodeKey == null && key == null) {
+                    currentNode.nodeValue = value;
+                    break;
+                } else {
+                    if (currentNode.next == null) {
+                        currentNode.next = new Node<>(key,value,null);
+                        elementsAmount++;
+                        break;
+                    } else {
+                        currentNode = currentNode.next;
+                    }
+                }
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        if (elementsAmount == 0) {
+            return null;
+        }
+        arrayIndex = (key == null ? 0 : Math.abs(key.hashCode() % valuesArray.length));
+        Node<K,V> currentNode = valuesArray[arrayIndex];
+        if (currentNode == null) {
+            return null;
+        }
+        while (true) {
+            if (currentNode.nodeKey != null && currentNode.nodeKey.equals(key)
+                    || currentNode.nodeKey == null && key == null) {
```

Comment:
you can use object equals
```suggestion
            if (currentNode.nodeKey != null && currentNode.nodeKey.equals(key)
                    || currentNode.nodeKey == key) {
```